cmd/ci-operator/main.go

```diff
@@ -398,13 +406,30 @@ func (o *options) Run() error {
 		secretName = o.sshSecret.Name
 	}
 	// load the graph from the configuration
-	buildSteps, postSteps, err := defaults.FromConfig(o.configSpec, o.jobSpec, o.templates, o.writeParams, o.artifactDir, o.promote, o.clusterConfig, o.targets.values, dryLogger, secretName)
+	buildSteps, postSteps, err := defaults.FromConfig(o.configSpec, o.jobSpec, o.templates, o.writeParams, o.artifactDir, o.promote, o.clusterConfig, o.leaseClient, o.targets.values, dryLogger, secretName)
 	if err != nil {
 		return fmt.Errorf("failed to generate steps from config: %v", err)
 	}
 
 	ctx, cancel := context.WithCancel(context.Background())
-
+	if o.leaseClient != nil {
+		defer func() {
```

Comment:
Isn't there some crazy on-exit code in ci-operator? We should use that

Reply:
This is a safety net in case the resources are not released normally (see `leaseStep.Run` and `leaseStep.release`). Not a good one, and arguably unnecessary, because the lease has timeouts anyway, but at least we get a message in the log.

In successful and unsuccessful executions, `defer` is sufficient.

I don't think `ci-operator`'s handling of interrupted executions allows us to do better (I had to discard some design choices because of that), see:

https://github.com/openshift/ci-tools/blob/d6ac3d9f96a4945c7c34381bc2cc1d6bc032ccdc/cmd/ci-operator/main.go#L408-L416

In case of an interrupt, all lease steps will execute a `release`, so we would be racing with them here (for 10 seconds, before `handler`'s `os.Exit(1)`).